src/repository.py

```diff
@@ -0,0 +1,221 @@
+from github.Repository import Repository
+
+from .constants import DOCUMENTATION_FOLDER_NAME
+from .exceptions import InputError, RepositoryClientError
+
+GITHUB_HOSTNAME = "github.com"
+HTTPS_URL_PATTERN = re.compile(rf"^https?:\/\/.*@?{GITHUB_HOSTNAME}\/(.+\/.+?)(.git)?$")
+ACTIONS_USER_NAME = "upload-charms-docs-bot"
+ACTIONS_USER_EMAIL = "upload-charms-docs-bot@users.noreply.github.com"
+ACTIONS_PULL_REQUEST_TITLE = "[upload-charm-docs] Migrate charm docs"
+ACTIONS_PULL_REQUEST_BODY = (
+    "This pull request was autogenerated by upload-charm-docs to migrate "
+    "existing documentation from server to the git repository."
+)
+PR_LINK_NO_CHANGE = "<not created due to no changes in repository>"
+
+CONFIG_USER_SECTION_NAME = "user"
+CONFIG_USER_NAME = (CONFIG_USER_SECTION_NAME, "name")
+CONFIG_USER_EMAIL = (CONFIG_USER_SECTION_NAME, "email")
+
+
+class Client:
+    """Wrapper for git/git-server related functionalities."""
+
+    def __init__(self, repository: Repo, github_repository: Repository) -> None:
+        """Construct.
+
+        Args:
+            repository: Client for interacting with local git repository.
+            github_repository: Client for interacting with remote github repository.
+        """
+        self._git_repo = repository
+        self._github_repo = github_repository
+        self._configure_git_user()
+
+    def _configure_git_user(self) -> None:
+        """Configure action git profile defaults.
+
+        Configured profile appears as the git committer.
+        """
+        config_reader = self._git_repo.config_reader(config_level="repository")
+        with self._git_repo.config_writer(config_level="repository") as config_writer:
+            if not config_reader.has_section(
+                CONFIG_USER_SECTION_NAME
+            ) or not config_reader.get_value(*CONFIG_USER_NAME):
+                config_writer.set_value(*CONFIG_USER_NAME, ACTIONS_USER_NAME)
+            if not config_reader.has_section(
+                CONFIG_USER_SECTION_NAME
+            ) or not config_reader.get_value(*CONFIG_USER_EMAIL):
+                config_writer.set_value(*CONFIG_USER_EMAIL, ACTIONS_USER_EMAIL)
+
+    def check_branch_exists(self, branch_name: str) -> bool:
+        """Check if branch exists on remote.
+
+        Args:
+            branch_name: Branch name to check on remote.
+
+        Raises:
+            RepositoryClientError: if unexpected error occurred during git operation.
+
+        Returns:
+            True if branch already exists, False otherwise.
+        """
+        try:
+            self._git_repo.git.fetch("origin", branch_name)
+            return True
+        except GitCommandError as exc:
+            if "couldn't find remote ref" in exc.stderr:
+                return False
+            raise RepositoryClientError(
+                f"Unexpected error checking existing branch. {exc=!r}"
+            ) from exc
+
+    def create_branch(self, branch_name: str, commit_msg: str) -> None:
```

Comment:
This method does more that creating the branch, it creates, commit and push.
I think it could be renamed

Reply:
Yeh the name isn't great, we can refactor it in another PR. The history is that it is intended to create the migration branch which is why it is doing the rest as well

Reply:
Ok